src/System.Private.Xml/src/System.Private.Xml.builds

```diff
@@ -3,6 +3,12 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <Project Include="System.Private.Xml.csproj" />
+    <Project Include="System.Private.Xml.csproj">
+      <TargetGroup>netcore50</TargetGroup>
+    </Project>
+    <Project Include="System.Private.Xml.csproj">
+      <TargetGroup>netcore50aot</TargetGroup>
+    </Project>
```

Comment:
No code from the the NET_NATIVE ifdef was removed, I only had to add a couple of whole classes back to the netcore50aot build, because of compile errors we were not seeing before in System.Private.Xml as we were not building for netcore50aot before. ( just notices I should've excluded one method  from netcore50aot build instead, so that's what I'm doing in my next commit)

All the different code lies in [XmlSerializer](https://github.com/dotnet/corefx/blob/dev/api/src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs) and [Compilation](https://github.com/dotnet/corefx/blob/dev/api/src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs). Looks like in netcore50aot we are eliminating use of a TempAssembly. I'm not sure if removing the netcore50 builds would make sense here.